sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/SimplifySizeExpressionCheck.kt

```diff
@@ -0,0 +1,234 @@
+/*
+ * SonarSource Kotlin
+ * Copyright (C) 2018-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.checks
+
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.lexer.KtToken
+import org.jetbrains.kotlin.lexer.KtTokens.ANDAND_Id
+import org.jetbrains.kotlin.lexer.KtTokens.EQEQ_Id
+import org.jetbrains.kotlin.lexer.KtTokens.EXCLEQ_Id
+import org.jetbrains.kotlin.lexer.KtTokens.EXCL_Id
+import org.jetbrains.kotlin.lexer.KtTokens.GT_Id
+import org.jetbrains.kotlin.lexer.KtTokens.LT_Id
+import org.jetbrains.kotlin.lexer.KtTokens.OROR_Id
+import org.jetbrains.kotlin.psi.KtBinaryExpression
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtConstantExpression
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.KtParenthesizedExpression
+import org.jetbrains.kotlin.psi.KtPrefixExpression
+import org.jetbrains.kotlin.psi.KtReferenceExpression
+import org.jetbrains.kotlin.psi.KtUnaryExpression
+import org.jetbrains.kotlin.psi.psiUtil.isNull
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.model.ResolvedCall
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.ArgumentMatcher
+import org.sonarsource.kotlin.api.CallAbstractCheck
+import org.sonarsource.kotlin.api.FunMatcher
+import org.sonarsource.kotlin.api.FunMatcherImpl
+import org.sonarsource.kotlin.api.fieldMatcher
+import org.sonarsource.kotlin.api.predictRuntimeIntValue
+import org.sonarsource.kotlin.plugin.KotlinFileContext
+
+private const val packageKotlinCollection = "kotlin.collections"
+private const val packageKotlinText = "kotlin.text"
+
+private const val interfaceKotlinCollection = "$packageKotlinCollection.Collection"
+private const val interfaceKotlinMap = "$packageKotlinCollection.Map"
+
+private val emptyArguments = listOf(emptyList<ArgumentMatcher>())
```

Comment:
Usually, we're using Upper snake case for top file final fields. We can maybe revisit it, but let's be consistent for now

Reply:
Fixed, but only for `private const` (we can talk about this if you think we should do it for all).